earth_enterprise/src/common/khCache.h

```diff
@@ -205,7 +205,7 @@ class khCache {
   }
   // calculates the current size of a cache item
   uint64 calculateCacheItemSize(Item *item) {
-    return khCacheItemSize + GetObjectSize(item->key) + GetObjectSize(item->val);
+    return khCacheItemSize + GetHeapUsage(item->key) + GetHeapUsage(item->val);
```

Comment:
I think you want to call `GetSize` here instead of `GetHeapUsage`.

Reply:
GetHeapSize has a specification for pointers/shared pointers that calls GetSize on the object pointed to. If I were to call GetSize with the current structure, it would call sizeof on the pointer/sharedpointer itself and then call GetHeapUsage on the pointer/sharedpointer. That would then detect the specific method to call and then call sizeof on the object pointed to and then GetHeapUsage again on the object pointed to. Since we calculate a cache items size excluding key and value, I figured this might not only be unnecessary but inaccurate. Not sure if I have the right idea.
We should call sizeof on the pointer itself right? Only the object pointed to?

Reply:
I think you're missing the actual size of the key and value here. This line:
`const uint64 khCacheItemSize = sizeof(khCacheItem<Key, Value>) - sizeof(Key) - sizeof(Value);`
takes out the size of the Key and Value, and this line:
`return khCacheItemSize + GetHeapUsage(item->key) + GetHeapUsage(item->val);`
doesn't add them back in. So for the Value in particular, which is a shared pointer to something, you have the size of the object it points to but not the size of the pointer itself. You could fix that on either line, either by not subtracting the key and value size when calculating khCacheItemSize or by calling GetSize here.